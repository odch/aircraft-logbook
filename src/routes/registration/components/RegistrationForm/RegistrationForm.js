import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import Button from '@material-ui/core/Button'
import FormControl from '@material-ui/core/FormControl'
import Input from '@material-ui/core/Input'
import InputLabel from '@material-ui/core/InputLabel'
import Typography from '@material-ui/core/Typography'
import withStyles from '@material-ui/core/styles/withStyles'

const styles = theme => ({
  failureMessage: {
    color: theme.palette.error.dark
  }
})

class RegistrationForm extends React.Component {
  handleEmailChange = e => {
    this.props.setEmail(e.target.value)
  }

  handlePasswordChange = e => {
    this.props.setPassword(e.target.value)
  }

  handleSubmit = e => {
    e.preventDefault()
    const { email, password } = this.props.registrationForm
    this.props.register(email, password)
  }

  render() {
    const {
      classes,
      registrationForm: { email, password, failed, submitted }
    } = this.props

    return (
      <React.Fragment>
        {failed && (
          <Typography className={classes.failureMessage}>
            <FormattedMessage id="registration.failed" />
          </Typography>
        )}
        <form onSubmit={this.handleSubmit}>
          <FormControl margin="normal" required fullWidth>
            <InputLabel htmlFor="email">
              <FormattedMessage id="registration.email" />
            </InputLabel>
            <Input
              id="email"
              name="email"
              autoComplete="email"
              onChange={this.handleEmailChange}
              value={email}
              disabled={submitted}
              data-cy="email"
              autoFocus
            />
          </FormControl>
          <FormControl margin="normal" required fullWidth>
            <InputLabel htmlFor="password">
              <FormattedMessage id="registration.password" />
            </InputLabel>
            <Input
              name="password"
              type="password"
              id="password"
              onChange={this.handlePasswordChange}
              value={password}
              disabled={submitted}
              data-cy="password"
            />
          </FormControl>
          <Button
            type="submit"
            fullWidth
            variant="raised"
            color="primary"
            disabled={submitted}
            data-cy="submit"
          >
            <FormattedMessage id="registration.register" />
          </Button>
        </form>
      </React.Fragment>
    )
  }
}

RegistrationForm.propTypes = {
  registrationForm: PropTypes.shape({
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired,
    failed: PropTypes.bool.isRequired,
    submitted: PropTypes.bool.isRequired
  }).isRequired,
  setEmail: PropTypes.func.isRequired,
  setPassword: PropTypes.func.isRequired,
  register: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(RegistrationForm)
