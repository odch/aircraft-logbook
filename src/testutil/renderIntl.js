import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
