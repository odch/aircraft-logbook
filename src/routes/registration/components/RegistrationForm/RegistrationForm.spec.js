import React from 'react'
import Input from '@material-ui/core/Input'
import renderIntl from '../../../../testutil/renderIntl'
import RegistrationForm from './RegistrationForm'

function renderRegistrationForm(
  registrationForm = {
    email: '',
    password: '',
    failed: false,
    submitted: false
  },
  setEmail = () => {},
  setPassword = () => {},
  register = () => {}
) {
  return renderIntl(
    <RegistrationForm
      registrationForm={registrationForm}
      setEmail={setEmail}
      setPassword={setPassword}
      register={register}
    />
  )
}

describe('components', () => {
  describe('RegistrationForm', () => {
    it('renders correctly', () => {
      const registrationFormData = {
        email: 'test@example.com',
        password: 'mypassword',
        failed: false,
        submitted: false
      }
      const tree = renderRegistrationForm(registrationFormData).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders error message if registration failed', () => {
      const registrationFormData = {
        email: 'test@example.com',
        password: '',
        failed: true,
        submitted: false
      }
      const tree = renderRegistrationForm(registrationFormData).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('calls setEmail function', () => {
      const setEmail = jest.fn()

      const component = renderRegistrationForm(undefined, setEmail)

      const usernameInput = component.root.find(
        el => el.type === Input && el.props.name === 'email'
      )
      usernameInput.props.onChange({
        target: { value: 'test@example.com' }
      })
      expect(setEmail).toHaveBeenCalledWith('test@example.com')
    })

    it('calls setPassword function', () => {
      const setPassword = jest.fn()

      const component = renderRegistrationForm(
        undefined,
        undefined,
        setPassword
      )

      const usernameInput = component.root.find(
        el => el.type === Input && el.props.name === 'password'
      )
      usernameInput.props.onChange({
        target: { value: 'mytestpassword' }
      })
      expect(setPassword).toHaveBeenCalledWith('mytestpassword')
    })

    it('calls register function on submit', () => {
      const registrationFormData = {
        email: 'test@example.com',
        password: 'mypassword',
        failed: false,
        submitted: false
      }
      const register = jest.fn()
      const e = {
        preventDefault: jest.fn().mockReturnThis()
      }

      const component = renderRegistrationForm(
        registrationFormData,
        undefined,
        undefined,
        register
      )

      const form = component.root.find(el => el.type === 'form')

      form.props.onSubmit(e)

      expect(e.preventDefault).toHaveBeenCalled()
      expect(register).toHaveBeenCalledWith('test@example.com', 'mypassword')
    })
  })
})
