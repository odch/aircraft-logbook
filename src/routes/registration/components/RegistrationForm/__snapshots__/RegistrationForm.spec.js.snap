// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components RegistrationForm renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
      data-shrink={true}
      htmlFor="firstname"
    >
      Vorname
      <span
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      data-cy="firstname"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={true}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="firstname"
        name="firstname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value="Max"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
      data-shrink={true}
      htmlFor="lastname"
    >
      Nachname
      <span
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      data-cy="lastname"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="lastname"
        name="lastname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value="Muster"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
      data-shrink={true}
      htmlFor="email"
    >
      E-Mail
      <span
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      data-cy="email"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="email"
        value="test@example.com"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
      data-shrink={true}
      htmlFor="password"
    >
      Passwort
      <span
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      data-cy="password"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="password"
        value="mypassword"
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
    data-cy="submit"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Registrieren
    </span>
  </button>
</form>
`;

exports[`components RegistrationForm renders error message if registration failed 1`] = `
Array [
  <p
    className="MuiTypography-root RegistrationForm-failureMessage-1 MuiTypography-body1"
  >
    Die Registrierung ist fehlgeschlagen.
  </p>,
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="firstname"
      >
        Vorname
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        data-cy="firstname"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={true}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="firstname"
          name="firstname"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value="Max"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="lastname"
      >
        Nachname
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        data-cy="lastname"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="lastname"
          name="lastname"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value="Muster"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="email"
      >
        E-Mail
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        data-cy="email"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
          value="test@example.com"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="password"
      >
        Passwort
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        data-cy="password"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      data-cy="submit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Registrieren
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </form>,
]
`;
